src/poetry/console/commands/add.py

```diff
@@ -236,7 +237,7 @@ def handle(self) -> int:
         if self.option("lock"):
             self._installer.lock()
 
-        self._installer.whitelist([r["name"] for r in requirements])
+        self._installer.whitelist([cast(str, r["name"]) for r in requirements])
```

Comment:
reached for a `cast` here as `requirements` is too loosely typed as `Dict[str, Union[str, List[str]]]`. planning to tighten that up after removing all ignored files

Reply:
That's kind of what I expected, sounds good.